src/java/main/org/apache/zookeeper/server/ServerStats.java

```diff
@@ -33,17 +34,17 @@
 public class ServerStats {
     private static final Logger LOG = LoggerFactory.getLogger(ServerStats.class);
 
-    private long packetsSent;
-    private long packetsReceived;
-    private long maxLatency;
-    private long minLatency = Long.MAX_VALUE;
-    private long totalLatency = 0;
-    private long count = 0;
+    private final AtomicLong packetsSent = new AtomicLong();
```

Comment:
Why are these 3 not part of `ServerMetrics`?
Is that a future step of the migration?

Reply:
It's being counted in requestLatency.

Reply:
I'm not sure if I see what you mean by "being counted"?
I've thought that these three metrics should be part of ServerMetrics as:
- packetsSent and packetsReceived as SimpleCounter,
- requestLatency as AvgMinMaxCounter

and should be referred as ServerMetrics.PACKETS_SENT, etc. like the rest.
Please let me know if didn't get it right.

Reply:
When making this change, we kept all existing metrics "as-is" and only added new metrics to `ServerMetrics`. The packet sent/receive and request_latency are both examples of metrics existing prior to this pull-request which are directly exposed in both in the `/metrics` admin command (where `ServerMetrics` metrics also report) as well as the `mntr` four letter command (where `ServerMetrics` metrics do not report).

The only changes to these existing metrics in this pull-request was converting some of them to `AtomicLong` for minor performance wins.

I'm happy to move both of these metrics to `ServerMetrics` if we want. I'm not sure if the names will 100% match the current values though. We'd also need to decide if we're happy losing these metrics in `mntr` or if I should port the existing `mntr` reporting logic to still query these from `ServerMetrics`.

This is also an open question for all other pre-existing metrics in `ServerStats`.

Reply:
Since `ServerMetrics` is a superset of `ServerStats` in terms of scope, we probably want to keep `ServerStats` as is and ultimately deprecate it in favor of `ServerMetrics`. I don't think there is a need to duplicate metrics in two places, which would be both a burden to maintain and a potential source of confusion.

Regarding reporting to `mntr`, we decided deprecate 4lw last year due to the limitation of its design, in particular around security, in favor of admin server endpoints  (`/metrics` in this case), so I don't there is a need to report newly added metrics to `mntr`. This also encourages users to migrate away from 4lw to admin end points.

Overall the state in current patch w.r.t this looks good to me.